download/download_test.go

```diff
@@ -652,6 +654,53 @@ func TestOneShotNotLongPollingSwitch(t *testing.T) {
 	}
 }
 
+func TestWarnOnNonBundleContentType(t *testing.T) {
+	ctx := context.Background()
+	fixture := newTestFixture(t)
+	fixture.server.bundles["not-a-bundle"] = bundle.Bundle{}
+
+	updates := make(chan *Update)
+
+	config := Config{}
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+
+	d := New(config, fixture.client, "/bundles/not-a-bundle").WithCallback(func(_ context.Context, u Update) {
+		updates <- &u
+	})
+	logger := test.New()
+	logger.SetLevel(logging.Debug)
+	d.logger = logger
+
+	d.Start(ctx)
+
+	// Give time for some download events to occur
+	time.Sleep(1 * time.Second)
```

Comment:
What does this give us over the `u1 := <-updates` below? I'd expect the channel read to block until something has is read (i.e. something has happened). If so, the sleep here would be redundant.

Reply:
This was copy-pasta from another test, and there seems to be a bunch of them following that convention. Without the sleep, these tests seem to block forever. So I guess you're right there's some race conditions going on in these tests 🙃

Reply:
Well if the race detection doesn't complain, neither will I. 😄

Reply:
Apparently, it _does_ [complain](https://github.com/open-policy-agent/opa/runs/5549284478?check_suite_focus=true) 😆 Except moving the stop call to earlier, it's not really apparent to me what would make this different from other similar tests here.

Reply:
This was fixed by removing code that wasn't necessary for these tests.